### PR TITLE
fix: allow canvas local config

### PR DESCRIPTION
**Pull request by pauldambra**
A user wants to be able to dynamically configure canvas recording... so, allow local config

### CODE REVIEW

File: src/extensions/replay/sessionrecording.ts [hunk 1]
```diff
@@ -303,15 +303,19 @@ export class SessionRecording {
...
-            ? {
-                  enabled: canvasRecording_server_side.enabled,
-                  fps: canvasRecording_server_side.fps,
-                  quality: canvasRecording_server_side.quality,
-              }
-            : undefined
+
+        const enabled = canvasRecording_client_side?.recordCanvas ?? canvasRecording_server_side?.enabled ?? false
+        const fps = canvasRecording_client_side?.canvasFps ?? canvasRecording_server_side?.fps ?? 0
+        const quality = canvasRecording_client_side?.canvasQuality ?? canvasRecording_server_side?.quality ?? 0
```
Comment (daibhin) [inline src/extensions/replay/sessionrecording.ts:312]: I would be careful about letting users arbitrarily set these two values. We could end up with very large recordings if they set the values too high, particularly the FPS. At the very least I would limit them to sensible ranges and emit a warning log

**Review by daibhin — CHANGES_REQUESTED** [review]

File: src/extensions/replay/sessionrecording.ts (hunk 1 above)
Comment (pauldambra) [reply src/extensions/replay/sessionrecording.ts:312]: ah... fair. 

have a feeling for reasonable range?

Comment (pauldambra) [reply src/extensions/replay/sessionrecording.ts:312]: fps... between 0 and 12?
quality between 0 and 1?

Comment (daibhin) [reply src/extensions/replay/sessionrecording.ts:312]: I was very conservative rolling it out initially 😂 

https://github.com/PostHog/posthog/blob/2a0a953fe91fb43d51f0f86fd0fae0a747db3017/posthog/api/decide.py#L373-L375

They sound reasonable. Unlikely many people will change the defaults. We can always dial it back in future SDK versions if it causes issues

**Review by daibhin — APPROVED** [review]
Nice 👌

**Pull request merged**